persistence/src/test/scala/hmda/persistence/processing/HmdaFileParserSpec.scala

```diff
@@ -3,40 +3,87 @@ package hmda.persistence.processing
 import java.time.Instant
 
 import akka.testkit.TestProbe
+import org.scalatest.BeforeAndAfterEach
 import com.typesafe.config.ConfigFactory
 import hmda.actor.test.ActorSpec
 import hmda.parser.fi.lar.LarCsvParser
+import hmda.parser.fi.ts.TsCsvParser
 import hmda.persistence.CommonMessages.GetState
-import hmda.persistence.processing.HmdaRawFile._
 import hmda.persistence.processing.HmdaFileParser._
+import hmda.persistence.processing.HmdaRawFile._
+
+import scala.concurrent.duration._
 
-class HmdaFileParserSpec extends ActorSpec {
+class HmdaFileParserSpec extends ActorSpec with BeforeAndAfterEach {
   import hmda.model.util.FITestData._
 
   val config = ConfigFactory.load()
 
   val submissionId = "12345-2017-1"
 
   val hmdaFileParser = createHmdaFileParser(system, submissionId)
-
   val probe = TestProbe()
 
   val timestamp = Instant.now.toEpochMilli
   val lines = fiCSV.split("\n")
   val badLines = fiCSVParseError.split("\n")
 
+  var initialSize: Int = _
+  var initalList: Seq[List[String]] = _
+
+  override def beforeEach(): Unit = {
```

Comment:
I really like the you are using the before each method to make the tests no longer dependent on each other. It seems like a really clean solution. I think it might clean the code up a bit more though to make the var be the hmdaFileParser and just create a new one of those before each test. It would make it so you don't have to carry around `initialSize` and `initialList`.